rhobs/README.md

```diff
@@ -26,7 +28,9 @@ creating a release branch as follows
 
 ```
 git fetch upstream --tags
-git push downstream v0.60.0:refs/heads/rhobs-rel-0.60.0-rhobs1
+git checkout v0.60.0
+git checkout -b rhobs-rel-0.60.0-rhobs1
+git push downstream rhobs-rel-0.60.0-rhobs1
```

Comment:
Although this accomplishes the one line that is being replaced, 
`git push downstream v0.60.0:refs/heads/rhobs-rel-0.60.0-rhobs1` is better since it pushes the tag from upstream (not local HEAD) to the remote downstream

Reply:
The one-liner is exactly what broke for me though.
From the commit message for this change:
```
I'm not 100% sure on the details but I think the push command
`git push downstream v0.60.0:refs/heads/rhobs-rel-0.60.0-rhobs1` will
only succeed if the remote branch exists already.
```

The `git help push` docu and especially the `<refspec>...` section has lots of details. The following excerpts are what led to my statement above in the commit message.
```
The <dst> tells which ref on the remote side is updated with this push. Arbitrary expressions cannot be used here, an actual ref must be named.
The refs/heads/* namespace will only accept commit objects, and updates only if they can be fast-forwarded.
All of the rules described above about what’s not allowed as an update can be overridden by adding an the optional leading + to a refspec (or using --force command line option). The only exception to this is that no amount of forcing will make the refs/heads/* namespace accept a non-commit object.
```

A new branch in `dst` is a non-commit object afaiu.

Reply:
> `git push downstream v0.60.0:refs/heads/rhobs-rel-0.60.0-rhobs1` will only succeed if the remote branch exists already.

That shouldn't be the case IIUC. This command is supposed to create a new branch and should have the remote HEAD set to `git ref-spec v0.60.0`. Did you forget to git fetch upstream by any chance? 

Could you please share that output of the push (`-v`) command?

Reply:
I think I understand the issue. This happens because we are trying to push a <sha> of type `tag` to a `branch`. So the solution is to dereference the `tag` to its corresponding `commit` when pushing. This can be achieved by using `<tag>^{commit}` syntax. E.g. 

`git push downstream +v0.60.0^{commit}:refs/heads/rhobs-rel-0.60.0-rhobs1` should work


This also pointed me to another issue in upstream prometheus that at tags are inconsistent. Some of them a lightweight instead of annotated. E.g. 

```
../upstream/prometheus-operator/prometheus-operator main ⇣
❯ git for-each-ref refs/tags | ag -A1 'v0\.61\.'
01465a8b3fd359f5e9b5dc06c488770bdacc3055 tag    refs/tags/pkg/apis/monitoring/v0.61.0
19917668abca594d6bd85ec4c1ec3a3cf14c7565 tag    refs/tags/pkg/apis/monitoring/v0.61.1
a2b88c27ad83bff45155ce97bedf6c2c47a41457 tag    refs/tags/pkg/apis/monitoring/v0.62.0
8f1db66a831359795efb1500866ee4f4507ddcd2 tag    refs/tags/pkg/client/v0.61.0
f2312dabd2ee373ca4529298c77d7fdd602a517f tag    refs/tags/pkg/client/v0.61.1
90664a67b1b66bb3051e4d2c8f0318b42986fa82 tag    refs/tags/pkg/client/v0.62.0
ba7ec03a08a81a9a40b128524d8104604a47baf9 commit refs/tags/v0.61.0     👈 light weight
a3c672fc95a37480e27c0cd50faebfb2ba8b986c tag    refs/tags/v0.61.1
ba3575645ee2c17b4050856f223e5e7d179cbdb9 tag    refs/tags/v0.62.0
```

Reply:
I suspect that the lightweight tags happen when the release shepherd forgot to sign the tag. Though it's the documented procedure, it's not enforced (not sure if it can be).

Reply:
AFAIK, you have to be explicit about `annotated` tags with git. I.e .. you have to use `git tag -a` otherwise, git creates light weight tags.

Reply:
I think that `git tag -s ...` also creates annotated tags.

Reply:
Indeed ! TIL :)